9805 numbers inside loan memo

emitche opened the pull request:
- Show table for numerical questions in print loan memo

Commit:
9805: Add initial table and describe logic for displaying table

Commit:
Merge remote-tracking branch 'origin/develop' into 9805_numbers_inside_loan_memo

Commit:
9805: Identify whether or not the previous question is numerical

Commit:
9805: Add styling to numerical question tables

Commit:
9805: Refine styling of numerical question tables

Commit:
9805: Remove logic that will not be used for numerical questions

Commit:
9805: Scope text question variables and add context in comments

Commit:
9805: Refine method for identifying numerical questions

Commit:
9805: Restore logic to hide unanswered and non applicable answers

Commit:
9805: Add contextual information for table

- Account for screen readers

Commit:
9805: Make borders be consistent for numerical display

Commit:
9805: Adjust spacing of questions

Commit:
9805: Add edit form label for numerical questions

Commit:
9805: Display table for numerical items in print view only

Commit:
9805: Address long question labels

Commit:
9805: Use translations for captiion text

Commit:
9805: Adjust CSS syntax

meltheadorable commented on app/models/loan_filtered_question.rb:
```diff
@@ -47,4 +47,8 @@ def visible?
     @response_set ||= loan.send(question_set.kind)
   end
+
+  def numerical?
```
I think you _might_ be able to use `has_number?` on the response object to control this rather than creating a new mapping here, it has a compatible value type.

emitche replied:
I'll look into `has_number?`.

meltheadorable commented on app/views/admin/loans/questionnaires/_questionnaire_group.html.slim:
```diff
@@ -84,19 +87,32 @@ div
-          - attrs = %w(class style src data-filename _moz_resizing href target frameborder width height)
 
           - if response.has_text?
+            / Display response as rich text. Specify which html is allowed.
```
is the rich text logic supposed to be part of this PR?

emitche replied:
No. That's an older feature from many months ago. I just added a comment for context. I can remove, if it is not needed.

emitche replied:
The `tags` and `attrs` were scoped outside of the if statement to which they belonged. I simply moved those into the if statement and tweaked the comments in the section.

meltheadorable commented on app/views/admin/loans/questionnaires/_questionnaire_group.html.slim:
```diff
@@ -145,5 +161,7 @@ div
           - if !response.group? && response.blank?
             .view-element.answer.blank = t("loan.no_answer")
 
+    - question.numerical? ? @previous_question_numerical = true : @previous_question_numerical = false
```
see the comment about `has_number?`

meltheadorable commented on app/views/admin/loans/questionnaires/_questionnaire_group.html.slim:
```diff
@@ -39,7 +39,10 @@ div
-          .question-label = question.full_number_and_label
+          // For numerical questions in print view, the question's label is displayed with
+          // the question's answer.
+          .question-label class=(question.numerical? ? 'hidden-print' : '')
```
see `has_number?`

meltheadorable reviewed (approved):
Looking good overall, but we might be able to get rid of some duplication here. 😄

Commit:
Merge remote-tracking branch 'origin/develop' into 9805_numbers_inside_loan_memo

Commit:
9805: Use existing method to distinguish numerical questions

Commit:
9805: Account for not answered and not applicable questions

Commit:
9805: Show consistent borders when questions are skipped

When number questions are skipped in print display, ensure the
borders are the same thickness.

Commit:
9805: Adjust logic to account for certain cases

Empty responses and non applicable questions will not show in
print loan memo. Adjust variable to display correct borders for
adjacent tables if a sibling question is not displayed.

emitche commented:
**QA Notes - Things That Were Changed**

In print loan memo, questions that have a number (percentage, currency, and number) are displayed in tables. Adjacent number reponses visually display as one larger table.

If a number question is marked not applicable or is empty, it should not display in the print loan memo. That question's absence in print loan memo should not break the flow of adjacent number reponses in print loan memo displaying as one combined table. To test this, for 3 consecutive number questions, answer the first and last. Do not answer the middle question or mark it as not applicable.

In the the Buisness Planning tab for a loan (such as http://localhost:3000/admin/loans/4/questions?filter=criteria), no change in the UI should be visible to the user in both the view and edit mode. A number question displays with its heading and either its response (view mode) or an input box (edit mode).

DevneyHamilton commented:
QA - works for me! (skipped qs s13 and 16)
<img width="582" alt="Screen St 2019-07-03 at 1 50 26 PM" src="https://user-images.githubusercontent.com/4730344/60613817-a5479600-9d99-11e9-9cfa-62eaa9eb5273.png">

Commit:
Merge remote-tracking branch 'origin/develop' into 9805_numbers_inside_loan_memo
